### PR TITLE
plug more leaks

### PR DESCRIPTION
**Pull request by christopherlam**

### CODE REVIEW

**Review by jralls — COMMENTED** [review]
Good. I'm happy that you're getting your head around memory management!

There's another leak at line 2192:
```
        filter_text = g_strdup (filter);
        filter_text = g_strdelimit (filter_text, ",",
                                    ';'); // make it conform to .gcm file list
        g_key_file_set_string (state_file, state_section, KEY_PAGE_FILTER,
                               filter_text);
        g_free (filter_text);
```
Leaks the first `filter_text`, plus `filter_text` is used in only in the if-block so it should be declared there instead of at function scope.

File: gnucash/gnome/gnc-plugin-page-register.c [hunk 1]
```diff
@@ -2150,7 +2148,7 @@ gnc_plugin_page_register_get_filter (GncPluginPage* plugin_page)
     GncPluginPageRegisterPrivate* priv;
     GNCLedgerDisplayType ledger_type;
     Account* leader;
-    const char* filter = NULL;
+    char* filter = NULL, *retval;
```
[the remark this answers is not in the document]
Comment (christopherlam) [reply gnucash/gnome/gnc-plugin-page-register.c:2151]: forgot to remove.

**Comment by Bob-IT** [comment]
> Leaks the first `filter_text`, plus `filter_text` is used in only in the if-block so it should be declared there instead of at function scope.

Doesn't g_strdelimit modify the 'filter_text' in place so does not need freeing till the end ?

**Comment by christopherlam** [comment]
All calls `g_strdelimit` are now modified to not use the return value, to make it more clearer it modifies the string argument.

**Pull request closed without merging**